src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,123 @@
+package duke.util;
+
+import duke.exception.DukeException;
+import duke.exception.InvalidCommandException;
+import duke.exception.NoDateProvidedException;
+import duke.exception.NoDescriptionException;
+
+import java.io.IOException;
+
+/**
+ * Parser
+ *
+ * <p>CS2103T AY19/20 Semester 2
+ * Individual project
+ * Duke project
+ *
+ * 30 Jan 2020
+ *
+ * @author Jel
+ */
+public class Parser {
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructs a Parser instance.
+     * @param tasks List where tasks are to be stored.
+     * @param ui User interface of the program.
+     */
+    public Parser(TaskList tasks, Ui ui) {
+        this.tasks = tasks;
+        this.ui = ui;
+    }
+
+    /**
+     * Parses user input.
+     * @param line The line of input to be parsed.
+     */
+    public void parseInput(String line) {
+        String trimmed = line.trim();
+        String cmd = getCommand(trimmed);
+        String descAndDate;
+
+        try {
+            switch (cmd) {
```

Comment:
I really liked the correct indentation of the switch case statement. Good job miss Lim